### ISSUE TITLE
The function of fully connected blocks

### ISSUE THREAD

**Issue by WindFall-wyx** [leesb7426/CVPR2022-Task-Discrepancy-Maximization-for-Fine-grained-Few-Shot-Classification#2]
Hi. Why use fully connected blocks to transform those two scores to two weights?  There are some other easy ways like softmax. How do you design this architecture?
Thanks a lot.

**Comment by leesb7426** [comment]
To construct the two weights (intra & inter), we considered two options.
1. Some channels, which are effective in the task, should get larger weight and otherwise get lower weight.
2. When channels well represent the self's object or are discriminative over a certain level, they should is given the maximum weight.

However, when we utilize a simple function like the softmax, option #2 cannot be satisfied. Therefore, we utilize the fully connected block for finding the certain level and adopt the activation function (tanh) for granting similar weight.

Thank you for your interest.

**Comment by WindFall-wyx** [comment]
Thanks for your reply.
I tried to train your model on one GTX2080Ti. But I found that it nearly took 11 hours to train, it shows that 35s/it. The GPU-Util sometimes changes to 0%.
Did I do something wrong? Is it normal？
![image](https://user-images.githubusercontent.com/69070640/190892991-330cc74b-5733-4f9d-8df2-6eb2092df7e1.png)
![image](https://user-images.githubusercontent.com/69070640/190893305-017e142a-4483-49b8-b085-ab125e8ac422.png)
